library/std/src/sys/windows/os.rs

```diff
@@ -321,21 +321,16 @@ pub fn temp_dir() -> PathBuf {
 #[cfg(not(target_vendor = "uwp"))]
 fn home_dir_crt() -> Option<PathBuf> {
     unsafe {
-        // The magic constant -4 can be used as the token passed to GetUserProfileDirectoryW below
-        // instead of us having to go through these multiple steps to get a token. However this is
-        // not implemented on Windows 7, only Windows 8 and up. When we drop support for Windows 7
-        // we can simplify this code. See #90144 for details.
-        use crate::sys::handle::Handle;
-
-        let me = c::GetCurrentProcess();
-        let mut token = ptr::null_mut();
-        if c::OpenProcessToken(me, c::TOKEN_READ, &mut token) == 0 {
-            return None;
-        }
-        let _handle = Handle::from_raw_handle(token);
+        // Defined in processthreadsapi.h.
+        const CURRENT_PROCESS_TOKEN: usize = -4isize as usize;
+
         super::fill_utf16_buf(
             |buf, mut sz| {
-                match c::GetUserProfileDirectoryW(token, buf, &mut sz) {
+                match c::GetUserProfileDirectoryW(
+                    ptr::from_exposed_addr_mut(CURRENT_PROCESS_TOKEN),
```

Comment:
This should use `ptr::invalid_mut` instead - we are using the pointer to smuggle an integer value, and not pointing to a pre-existing exposed allocation.

Reply:
That makes sense, I was misled by the suggestion when I tried to just do an as cast, not too familiar with the new ptr apis.